lib/grppcatr.gi

```diff
@@ -466,6 +466,49 @@ end );
 
 #############################################################################
 ##
+#M  MaximalNormalSubgroups( <G> )
+##
+InstallMethod( MaximalNormalSubgroups, "for solvable groups",
+              [ IsGroup ],
+              RankFilter( IsGroup and IsSolvableGroup )
+              - RankFilter( IsGroup ),
+
+  function( G )
+    local Gf,     # FactorGroup of G
+          hom,    # homomorphism from G to Gf
+          MaxGf;  # MaximalNormalSubgroups of Gf
+
+    if 0 in AbelianInvariants(G) then
+      # (p) is a maximal normal subgroup in Z for every prime p
+      Error("number of maximal normal subgroups is infinity");
+    elif IsAbelian(G) then
+      if not IsPcGroup(G) then
+        # convert it to an Abelian PcGroup with same invariants
+        Gf := AbelianGroup(IsPcGroup, AbelianInvariants(G));
+        hom := IsomorphismGroups(G, Gf);
```

Comment:
The whole purpose of `CanEasilyComputePcgs` *is* that a group might know to be solvable (say because a flag has been set by the user) but there still is no good method for computing a PCGS. FpGroup are exactly of this form, unless an isomorphism to a solvable group has already been constructed.

It is not a good idea to use expensive functions (such as potentially `IsomorphismGroups`) just because the context is not clear (and no -- this is not really what the NiceMonomorophisms are intended for).
For FpGroups there is for example `MaximalAbelianQuotient` which is probably as efficient as one could hope for.

I'm slightly confused by NormalMaximalSubgroups. Aren't we talking about MaximalNormalSubgroups?